packages/@lwc/engine-core/src/framework/update-component-value.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { VM } from './vm';
+import { componentValueMutated } from './mutation-tracker';
+
+export function updateComponentValue(vm: VM, key: string, newValue: any) {
+    const { cmpFields } = vm;
+    if (newValue !== cmpFields[key]) {
+        cmpFields[key] = newValue;
+
+        componentValueMutated(vm, key);
```

Comment:
The other place where `componentValueMutated` is invoked is in [decorators/api.ts](https://github.com/salesforce/lwc/blob/4b8b03e0b4246205984a020c4fb59e92a18ad4ac/packages/%40lwc/engine-core/src/framework/decorators/api.ts#L69-L71). However, we don't check there if the value is the same or not. Do you know why we don't do a similar check?